crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -142,25 +144,35 @@ impl ModuleSubscriptions {
             .set(subscriptions.num_queries() as i64);
     }
 
-    /// Broadcast a ModuleEvent to all interested subscribers.
-    ///
-    /// It's recommended to take a read lock on `subscriptions` field *before* you commit
-    /// the transaction that will give you the event you pass here, to prevent a race condition
-    /// where a just-added subscriber receives the same update twice.
-    pub async fn broadcast_event(
+    /// Commit a transaction and broadcast its ModuleEvent to all interested subscribers.
+    pub fn commit_and_broadcast_event(
         &self,
         client: Option<&ClientConnectionSender>,
-        subscriptions: &SubscriptionManager,
-        event: Arc<ModuleEvent>,
-    ) {
-        match event.status {
-            EventStatus::Committed(_) => {
-                tokio::task::block_in_place(|| self.broadcast_commit_event(subscriptions, event, client))
+        mut event: ModuleEvent,
+        ctx: &ExecutionContext,
+        tx: MutTx,
+    ) -> Result<Result<Arc<ModuleEvent>, WriteSkew>, DBError> {
+        let subscriptions = self.subscriptions.read();
```

Comment:
```suggestion
       // Take a read lock on `subscriptions` before committing tx
       // else it can result in subscriber receiving duplicate updates.
        let subscriptions = self.subscriptions.read();
```
Will prefer to not still not drop this commment.